style: tightened up trending content for smaller screens

jtucholski opened the pull request:
## Description of the change

This PR tries to tighten up some of the cards used on the Trending Content screen when viewed on regular laptop screens. The existing spacing was left untouched for `xl` screens. 

- **Related issues**: closes #674 

## Screenshots (laptop screen)

### Before
![127 0 0 1_trending-content-laptop-before](https://github.com/user-attachments/assets/929df1f9-f35e-4487-aaf4-1fef42e0c3f1)

### After
![127 0 0 1_trending-content-laptop-after](https://github.com/user-attachments/assets/3f4fc416-8b96-4605-8588-6b6913c78934)

## Screenshots (large monitor)

### Before
![127 0 0 1_trending-content-monitor-before](https://github.com/user-attachments/assets/80fdd404-1101-4bb8-8f81-63eb64110e25)

### After
![127 0 0 1_trending-content-monitor-after](https://github.com/user-attachments/assets/91a99260-52fa-415f-a5e9-6ffaf2a8f8af)

carddev81 commented on frontend/src/Pages/StudentLayer1.tsx:
Just a note about when there are more than four helpful links the cards get dispalyed in a single column rather than in four columns. Just confirming this is correct behavior.

>4 helpful links
![image](https://github.com/user-attachments/assets/882e1b5d-3203-4433-9009-bb5eff92dd55)

<=4 helpful links
![image](https://github.com/user-attachments/assets/072f8302-7695-4d85-96c0-0d7b90e9d0fa)

jtucholski replied:
Thanks @carddev81 -- I'll look into this and see if I can make it behave like the "See More" in the Featured Content section.

carddev81 reviewed (approved):
Added a note to the PR, nothing changed to cause this, I just noticed this while testing.

calisio reviewed (approved):
I think it looks good. Interestingly, my computer is an xl layout, so I only see this when I make my screen smaller. Once Rich's comment is fixed it's good to merge!

jtucholski commented:
@calisio @carddev81 I've added an ExpandableCardGrid component and applied it to the Helpful Links section seen in Trending Content. The same component was applied wherever we had Featured Content or Favorite Libraries visible.

calisio commented:
Tried it out and it looks good- seems like there are a few conflicts but once those are fixed its good to merge

carddev81 reviewed (approved):
Looks Great!

jtucholski commented:
@carddev81 Something weird is going on with this conflict. It says the issue is in ExpandableCardGrid but that's because the changes took place to `FeaturedContent` which I removed and refactored into an ExpandableCardGrid component.

It looks like `FeaturedContent` was updated to include the Search modals. Is that something that needs to exist in the `FeaturedContent` component?
